### PR TITLE
Clean up old warnings in F.interpolate

### LINKED ISSUE #71720: Old warnings from F.interpolate docs should be cleaned up
## Issue description

The [`F.interpolate()` docs](https://pytorch.org/docs/master/generated/torch.nn.functional.interpolate.html) include a couple of old warnings regarding the behavior of `align_corners` and `recompute_scale_factor` referencing PyTorch v0.3.1 and v1.6.0, respectively. These should be removed as no longer relevant. Also, somewhat related: the docstring for `recompute_scale_factor` does not specify what the default value for that arg is and should be updated as well.

cc @brianjo @mruberry @albanD @jbschlosser @walterddr @kshitij12345

### PR DESCRIPTION
**Pull request by pejato**
Fixes #71720

This PR removes the old warnings for `recompute_scale_factor` and `align_corners`. 

Looking at this, I realize that the tests I modified don't really catch whether or not a warning is created for  `recompute_scale_factor`. If desired, I can add a couple lines into the tests there to pass a floating point in the `scale_factors` kwarg, along with `recompute_scale_factor=None`.

Let me know how this looks, thanks so much!

### CODE REVIEW

**Comment by pejato** [comment]
Shame on me for not running the whole test suite locally.

I am actually a bit confused here as to why `recompute_scale_factor` is an `Optional[bool]` defaulting to `None`. At a glance it seems like we should just be able to replace it with a plain old `bool` defaulting to `False` but I get a bunch of test failures elsewhere when I run `python test/test_jit.py` with that so I am unsure..

**Comment by albanD** [comment]
> Shame on me for not running the whole test suite locally.

Definitely not your fault! Changing the default to False would definitely be the right thing to do. But as you've seen it is significantly more dangerous than just removing warnings.
So let's only remove the warning here (as you've done in the second commit) and we can revisit changing the default in a future PR if you want to do that as well.

**Review by albanD — APPROVED** [review]
Perfect!
Thanks for the fix!

**Comment by pejato** [comment]
> Perfect!
> 
> Thanks for the fix!

Thanks for the review!

**Comment by albanD** [comment]
I'll take it from here. This PR will be closed once the commit is in master!

**Review by jbschlosser — APPROVED** [review]
LGTM :) Thanks for the fix!

**Comment by pejato** [comment]
Thanks for the labels @jbschlosser ^^ I'll make sure to add those next time.
